test/functional/wallet_dump.py

```diff
@@ -125,18 +129,18 @@ def run_test (self):
         self.nodes[0].walletpassphrase('test', 10)
         # Should be a no-op:
         self.nodes[0].keypoolrefill()
-        self.nodes[0].dumpwallet(tmpdir + "/node0/wallet.encrypted.dump")
+        self.nodes[0].dumpwallet(wallet_enc_dump)
 
         found_addr, found_script_addr, found_addr_chg, found_addr_rsv, _, witness_addr_ret = \
-            read_dump(tmpdir + "/node0/wallet.encrypted.dump", addrs, script_addrs, hd_master_addr_unenc)
+            read_dump(wallet_enc_dump, addrs, script_addrs, hd_master_addr_unenc)
         assert_equal(found_addr, test_addr_count)
         assert_equal(found_script_addr, 2)
         assert_equal(found_addr_chg, 90*2 + 50)  # old reserve keys are marked as change now
         assert_equal(found_addr_rsv, 90*2) 
         assert_equal(witness_addr_ret, witness_addr)
 
         # Overwriting should fail
-        assert_raises_rpc_error(-8, "already exists", self.nodes[0].dumpwallet, tmpdir + "/node0/wallet.unencrypted.dump")
+        assert_raises_rpc_error(-8, "already exists", lambda: self.nodes[0].dumpwallet(wallet_enc_dump))
```

Comment:
should be `wallet_unenc_dump`

Reply:
Both exists, so it shouldn't matter when we are just testing that overwriting fails.

Also, the last calls to node 0 are to encrypt and then dump the encrypted wallet. So imo it is easier to understand the code (with less context) if we try to overwrite the dump of the encrypted wallet.